Lib/test/test_tools/test_makefile.py

```diff
@@ -0,0 +1,66 @@
+"""
+Tests for `Makefile`.
+"""
+
+import os
+import unittest
+from test import support
+
+MAKEFILE = os.path.join(support.REPO_ROOT, 'Makefile')
+
+if not support.check_impl_detail(cpython=True):
+    raise unittest.SkipTest('cpython only')
+if support.MS_WINDOWS:
+    raise unittest.SkipTest('Windows does not support make')
+if not os.path.exists(MAKEFILE) or not os.path.isfile(MAKEFILE):
+    raise unittest.SkipTest('Makefile could not be found')
```

Comment:
Why skip the test here rather than failing the test?

Reply:
I open to fail it, if all platforms and use-cases do require the `Makefile`. I just assume that sometimes it can be missing.

Reply:
Yeah... I worry that the location of this test itself could change, which might lead to this test silently being skipped :) but I can definitely see arguments both ways

Reply:
> the location of this test itself could change

It does not use any relative paths at all, only absolute ones defined in `test.support`. It should not be sensible to the test exact location.

Reply:
Then I guess the only worry would be if we deleted the Makefile altogether and moved to a different build system. That would also lead to this test being silently skipped... but that's very speculative. The way you have it now should be fine :)